pkg/yurthub/tenant/tenant.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2020 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package tenant
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"github.com/openyurtio/openyurt/cmd/yurthub/app/config"
+	"github.com/openyurtio/openyurt/pkg/yurthub/cachemanager"
+	"github.com/openyurtio/openyurt/pkg/yurthub/kubernetes/rest"
+	"github.com/openyurtio/openyurt/pkg/yurthub/util"
+)
+
+type TenantManager struct {
+	Client *kubernetes.Clientset
+
+	Informer cache.SharedIndexInformer
+
+	Storage cachemanager.StorageWrapper
+
+	TenantSecret *v1.Secret
+}
+
+func New(cfg *config.YurtHubConfiguration, configMgr *rest.RestConfigManager) *TenantManager {
+
+	tenantNs := ParseTenantNs(cfg)
+	klog.Infof("parse tenant ns: %s", tenantNs)
+	if tenantNs == "" {
+		return nil
+	}
+
+	tenantMgr := &TenantManager{}
+
+	kubeConf := configMgr.GetRestConfig(true)
+	clientSet, err := kubernetes.NewForConfig(kubeConf)
+	if err != nil {
+
+		klog.Errorf("failed to create client set, %v", err)
+
+	}
+
+	tenantMgr.Storage = cfg.StorageWrapper
+
+	if clientSet != nil {
+
+		factory := informers.NewSharedInformerFactoryWithOptions(clientSet, time.Minute, informers.WithNamespace(tenantNs))
```

Comment:
please reuse `sharedInformer` in `config.YurtHubConfiguration` instead of creating a new `sharedInformer`.